### PR TITLE
PLAT-64690: Disable scroll animation with 5-way navigation

### PR DESCRIPTION
**Pull request by yeram**
### Checklist

* [x] I have read and understand the [contribution guide](http://enactjs.com/docs/developer-guide/contributing/)
* [x] A [CHANGELOG entry](http://enactjs.com/docs/developer-guide/contributing/changelogs/) is included
* [ ] At least one test case is included for this feature or bug fix
* [x] Documentation was added or is not needed

* [ ] This is an API breaking change

### Issue Resolved / Feature Added
[//]: # (Describe the issue resolved or feature added by this pull request)
Fix stuttering issue by disabling animation with 5-way navigation

### Resolution
[//]: # (Does the code work as intended?)
[//]: # (What is the impact of this change and *why* was it made?)
Allow `animate` as a prop and use it to branch whether it animate or not

### Additional Considerations
[//]: # (How should the change be tested?)
[//]: # (Are there any outstanding questions?)
[//]: # (Were any side-effects caused by the change?)
Not pass to `ui/VirtualListBase` at this time since it doesn't have any logic relate to animation

### Links
[//]: # (Related issues, references)
PLAT-64690

### Comments
Enact-DCO-1.0-Signed-off-by: Yeram Choi yeram.choi@lge.com

### CODE REVIEW

File: packages/ui/Scrollable/ScrollableNative.js [hunk 1]
```diff
@@ -755,7 +755,7 @@ class ScrollableBaseNative extends Component {
 
 		this.lastInputType = 'pageKey';
 
-		this.scrollToAccumulatedTarget(pageDistance, canScrollVertically, this.props.overscrollEffectOn.pageKey);
+		this.scrollToAccumulatedTarget(pageDistance, canScrollVertically, this.props.overscrollEffectOn.pageKey, this.props.animate);
```
Comment (yeram) [inline packages/ui/Scrollable/ScrollableNative.js]: I saw animation while paging feature validation. I pass `animate` here to disable animation explicitly.

File: packages/moonstone/VirtualList/VirtualListBase.js [hunk 2]
```diff
@@ -572,7 +581,7 @@ const VirtualListBaseFactory = (type) => {
 							// So we find nextIndex only when currentExtent is different from candidateExtent.
 							if (currentExtent !== candidateExtent) {
 								nextIndex = this.findNearestSpottableItemInExtent(currentIndex, candidateExtent);
-								animate = (wrap === true);
+								animate = this.props.animate && (wrap === true);
```
Comment (yeram) [inline packages/moonstone/VirtualList/VirtualListBase.js]: Not sure what this line means. But I added checking `animate` to disable it anyway.

File: packages/moonstone/Scrollable/Scrollable.js [hunk 3]
```diff
@@ -452,7 +461,7 @@ class ScrollableBase extends Component {
 			ev.preventDefault();
 		}
 
-		this.animateOnFocus = true;
+		this.animateOnFocus = this.props.animate;
```
Comment (yeram) [inline packages/moonstone/Scrollable/Scrollable.js]: There is no initial value of `animateOnFocus`. I guess it's safer to assign value, but leave it as-is for now.

**Comment by ryanjduffy** [comment]
In order to isolate the behavior change to moonstone, I've updated the `ui` components to use `noAnimation` instead of `animate` so that the default behavior is still animated scrolling for non-moonstone targets. I've left the prop as `private` because I expect that we'll attempt to improve the performance so that we can re-enable animation by default without altering public APIs.

File: packages/moonstone/CHANGELOG.md [hunk 4]
```diff
@@ -7,6 +7,7 @@ The following is a curated list of changes in the Enact moonstone module, newest
 ### Fixed
 
 - `moonstone/VideoPlayer` to disable pointer mode when hiding media controls via 5-way
+- `moonstone/VirtualList` and `moonstone/Scrollable` to not to animate with 5-way navigation by default
```
Comment (webOS101) [inline packages/moonstone/CHANGELOG.md]: `moonstone/Scrollable` is private, so we don't discuss it in the CHANGELOG.

**Pull request merged**